### ISSUE TITLE
playpause() does not work

### ISSUE THREAD

**Issue by schrock** [regosen/Gapless-5#31]
**Describe the bug**
Calling `playpause()` does not pause the player. In order to workaround this, I had to implement my own logic. Note that I have disabled Web Audio.

```
var playerPaused = true;

player.onplay = function () {
    playerPaused = false;
}
player.onpause = function () {
    playerPaused = true;
}

function audioPause() {
    if (playerPaused) {
        player.play();
    } else {
        player.pause();
    }
}
```

**Debug output**
There was no console output when enabling debugging and calling playpause().

**Comment by schrock** [comment]
The underlying problem is that the internal play/pause logic depends on the built-in UI. I am not using the built-in UI.

**Comment by regosen** [comment]
Thanks for finding all these issues!  I'm on vacation at the moment, but I can look at PRs or take a closer look next week.

**Comment by regosen** [comment]
Wow, indeed this is a nasty bug!  I just pushed the fix and published a new version to npm (1.3.12) containing this fix, plus your recent PRs.  Thanks again for all your reporting.

**Comment by schrock** [comment]
The last commit you made fixed `playpause()`, but it seems to have broken `cue()`. When I call `cue()`, it resets the reported player position, but does not actually affect playback in any way. No log messages are output on the call to `cue()`.

**Comment by regosen** [comment]
Yikes, I see how it happened now.  It was caused by really old unused parameters having unintended behavior.  Ok, I just published a fix (1.3.13), please try it now!

**Comment by schrock** [comment]
That `cue()` issue is fixed, but the latest update has caused another problem. When I click the next button in my app, I call the following function. The intended behavior is to always start playing the next track, even if the playback was previously paused or stopped. 
```
function audioNext() {
	player.next();
	player.cue();
}
```
This is why I call `cue()` after `next()`. This works as intended if playback is paused or stopped, but if I click the next button while playing, it just starts playing from the beginning of the current track.

```
Setting starting track to 0
gapless5.js:619 Loadable indices: [0,1]
gapless5.js:631 Loading track 0: /play?path=%2Fmovie%20ost%2FHook%2F01%20Prologue.flac&duration=90.293333&track_index=0
gapless5.js:631 Loading track 1: /play?path=%2Fmovie%20ost%2FHook%2F02%20We%20Don't%20Wanna%20Grow%20Up.flac&duration=110.133333&track_index=0
gapless5.js:260 Loading /play?path=%2Fmovie%20ost%2FHook%2F01%20Prologue.flac&duration=90.293333&track_index=0
gapless5.js:233 Playing HTML5 Audio: /play?path=%2Fmovie%20ost%2FHook%2F01%20Prologue.flac&duration=90.293333&track_index=0
gapless5.js:185 onEnded() will be called in 90.28 sec
```
Clicking the next button...
```
Setting track number to 1
gapless5.js:619 Loadable indices: [1,2]
gapless5.js:158 Stopping: /play?path=%2Fmovie%20ost%2FHook%2F01%20Prologue.flac&duration=90.293333&track_index=0
gapless5.js:635 Unloaded track 0: /play?path=%2Fmovie%20ost%2FHook%2F01%20Prologue.flac&duration=90.293333&track_index=0
gapless5.js:631 Loading track 2: /play?path=%2Fmovie%20ost%2FHook%2F03%20Banning%20Back%20Home.flac&duration=142.173333&track_index=0
gapless5.js:463 Setting track number to 0
gapless5.js:619 Loadable indices: [0,1]
gapless5.js:631 Loading track 0: /play?path=%2Fmovie%20ost%2FHook%2F01%20Prologue.flac&duration=90.293333&track_index=0
gapless5.js:158 Stopping: /play?path=%2Fmovie%20ost%2FHook%2F03%20Banning%20Back%20Home.flac&duration=142.173333&track_index=0
gapless5.js:635 Unloaded track 2: /play?path=%2Fmovie%20ost%2FHook%2F03%20Banning%20Back%20Home.flac&duration=142.173333&track_index=0
gapless5.js:158 Stopping: /play?path=%2Fmovie%20ost%2FHook%2F02%20We%20Don't%20Wanna%20Grow%20Up.flac&duration=110.133333&track_index=0
gapless5.js:260 Loading /play?path=%2Fmovie%20ost%2FHook%2F01%20Prologue.flac&duration=90.293333&track_index=0
gapless5.js:233 Playing HTML5 Audio: /play?path=%2Fmovie%20ost%2FHook%2F01%20Prologue.flac&duration=90.293333&track_index=0
gapless5.js:185 onEnded() will be called in 90.28 sec
```

**Comment by regosen** [comment]
I don't see how this could be caused by the latest update, which that didn't touch `next()` or anything that `next()` calls (but maybe it was still a recent update before that).  Looks like you have loadLimit set to 2, just like me, but comparing your console output to mine, I see a key difference:
- your tracks are still playing with HTML5 Audio when you call next(), while mine have already finished loading HTML5 Audio and switched to WebAudio by that time.

So I wonder if there's something going on with the fact that it's still trying to load WebAudio by the time you call next().  Can you try something for me?  
- initialize Gapless5 with `useHTML5Audio: false`

Then let's see if the issue still happens.  It might take a while to finish loading your tracks for playback though (I see that you're using 90-second FLAC files), you might want to test with something smaller.

```
gapless5.js:203 Playing WebAudio: psounds/Synth1.wav
gapless5.js:185 onEnded() will be called in 3.55 sec

gapless5.js:463 Setting track number to 1
gapless5.js:619 Loadable indices: [1,2]
gapless5.js:158 Stopping: psounds/Synth1.wav
gapless5.js:635 Unloaded track 0: psounds/Synth1.wav
gapless5.js:631 Loading track 2: psounds/Synth4.wav
gapless5.js:203 Playing WebAudio: psounds/Synth2.wav
gapless5.js:185 onEnded() will be called in 1.65 sec
```

**Comment by schrock** [comment]
I think the issue is related to the call to `cue()` rather than `next()`. If I remove the call to `cue()`, then it does properly move on the next track. I would still like to call `cue()` to start playing if playback was paused or stopped, though. Additionally, I was only using HTML5 audio in my tests, not WebAudio.

Here are the results of using only Web Audio.
`player = new Gapless5({ logLevel: LogLevel.Debug, useHTML5Audio: false, loop: true, loadLimit: 2, tracks: tracks, startingTrack: position });`

```
Setting starting track to 0
gapless5.js:619 Loadable indices: [0,1]
gapless5.js:631 Loading track 0: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F01%20Opera%20Singer.mp3&duration=246.616667&track_index=0
gapless5.js:631 Loading track 1: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F02%20Meanwhile%2C%20Rick%20Jame.....mp3&duration=237.395917&track_index=0
gapless5.js:260 Loading /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F01%20Opera%20Singer.mp3&duration=246.616667&track_index=0
gapless5.js:203 Playing WebAudio: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F01%20Opera%20Singer.mp3&duration=246.616667&track_index=0
gapless5.js:185 onEnded() will be called in 246.60 sec
```
Clicking the next button...
```
Setting track number to 1
gapless5.js:619 Loadable indices: [1,2]
gapless5.js:158 Stopping: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F01%20Opera%20Singer.mp3&duration=246.616667&track_index=0
gapless5.js:635 Unloaded track 0: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F01%20Opera%20Singer.mp3&duration=246.616667&track_index=0
gapless5.js:631 Loading track 2: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F03%20Shadow%20Stabbing.mp3&duration=187.318833&track_index=0
gapless5.js:463 Setting track number to 0
gapless5.js:619 Loadable indices: [0,1]
gapless5.js:631 Loading track 0: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F01%20Opera%20Singer.mp3&duration=246.616667&track_index=0
gapless5.js:158 Stopping: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F03%20Shadow%20Stabbing.mp3&duration=187.318833&track_index=0
gapless5.js:635 Unloaded track 2: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F03%20Shadow%20Stabbing.mp3&duration=187.318833&track_index=0
gapless5.js:158 Stopping: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F02%20Meanwhile%2C%20Rick%20Jame.....mp3&duration=237.395917&track_index=0
gapless5.js:260 Loading /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F01%20Opera%20Singer.mp3&duration=246.616667&track_index=0
gapless5.js:220 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'stop')
    at gapless5.js:220:18
(anonymous) @ gapless5.js:220
Promise.then (async)
playAudioFile @ gapless5.js:201
Gapless5Source.play @ gapless5.js:263
Gapless5FileList.gotoTrack @ gapless5.js:480
Gapless5.gotoTrack @ gapless5.js:1103
Gapless5.next @ gapless5.js:1172
audioNext @ main.js:413
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
gapless5.js:203 Playing WebAudio: /play?path=%2Fnormal%2FCake%2FComfort%20Eagle%2F01%20Opera%20Singer.mp3&duration=246.616667&track_index=0
gapless5.js:185 onEnded() will be called in 246.60 sec
```

**Comment by regosen** [comment]
Oh I see.  It might be a special condition of calling cue() right after next().  Try replacing the `this.cue` implementation with this:

```
  this.cue = () => {
    if (this.currentPosition() > 0) {
      this.prev(true);
    }
    this.play();
  };
  ```
  
  Does that help?

**Comment by schrock** [comment]
That fixes it. It works using both HTML5 audio and WebAudio.

**Comment by regosen** [comment]
Ok great!   I just pushed this fix to version 1.3.14
